packages/kit/src/runtime/client/utils.js

```diff
@@ -23,72 +24,135 @@ export function scroll_state() {
 	};
 }
 
-/** @param {Event} event */
-export function find_anchor(event) {
+const warned = new WeakSet();
+
+/** @typedef {keyof typeof valid_link_options} LinkOptionName */
+
+const valid_link_options = /** @type {const} */ ({
+	'preload-code': ['', 'off', 'tap', 'hover', 'viewport', 'eager'],
+	'preload-data': ['', 'off', 'tap', 'hover'],
+	noscroll: ['', 'off'],
+	reload: ['', 'off']
+});
+
+/**
+ * @template {LinkOptionName} T
+ * @param {Element} element
+ * @param {T} name
+ */
+function link_option(element, name) {
+	const value = /** @type {typeof valid_link_options[T][number] | null} */ (
+		element.getAttribute(`data-sveltekit-${name}`)
+	);
+
+	return __SVELTEKIT_DEV__ ? validate_link_option(element, name, value) : value;
+}
+
+/**
+ * @template {LinkOptionName} T
+ * @template {typeof valid_link_options[T][number] | null} U
+ * @param {Element} element
+ * @param {T} name
+ * @param {U} value
+ */
+function validate_link_option(element, name, value) {
+	if (warned.has(element) || value === null) return /** @type {U} */ (null);
+
+	// @ts-expect-error - includes is dumb
+	if (!valid_link_options[name].includes(value)) {
+		console.error(
```

Comment:
```suggestion
function validate_link_option(element, name, value) {
	if (value === null) return /** @type {U} */ (null);

	// @ts-expect-error - includes is dumb
	if (!valid_link_options[name].includes(value) && !warned.has(element)) {
		console.error(
```

just noticed that the value might change after a warning was logged

on second thought, this function might be better as a side-effect-y one instead of one that returns a value